dcos/auth.py

```diff
@@ -0,0 +1,308 @@
+        config.set_val("core.dcos_acs_token", token)
+
+    return token
+
+
+def _prompt_for_uid_password(username, hostname):
+    """Get username/password for auth
+
+    :param username: username user for authentication
+    :type username: str
+    :param hostname: hostname for credentials
+    :type hostname: str
+    :returns: username, password
+    :rtype: str, str
+    """
+
+    if username is None:
+        sys.stdout.write("{}'s username: ".format(hostname))
+        sys.stdout.flush()
+        username = sys.stdin.readline().strip()
+
+    password = getpass.getpass("{}@{}'s password: ".format(username, hostname))
+
+    return username, password
+
+
+def _get_dcostoken_by_dcos_uid_password_auth(
+        dcos_url, username=None, password=None):
+    """
+    Get DC/OS Authentication Token by DC/OS uid password auth
+
+    :param dcos_url: url to cluster
+    :type dcos_url: str
+    :param username: username to auth with
+    :type username: str
+    :param password: password to auth with
+    :type password: str
+    :returns: DC/OS Authentication Token if successful login
+    :rtype: str
+    """
+
+    url = urlparse(dcos_url)
+    hostname = url.hostname
+    username = username or url.username
+    password = password or url.password
+
+    if password is None:
+        username, password = _prompt_for_uid_password(username, hostname)
+
+    creds = {"uid": username, "password": password}
+    return _get_dcostoken_by_post_with_creds(dcos_url, creds)
+
+
+def dcos_uid_password_auth(dcos_url, username=None, password=None):
+    """
+    Authenticate user using DC/OS uid password auth
+
+    Raises exception if authentication fails.
+
+    :param dcos_url: url to cluster
+    :type dcos_url: str
+    :param username: username to auth with
+    :type username: str
+    :param password: password to auth with
+    :type password: str
+    :rtype: None
+    """
+
+    dcos_token = _get_dcostoken_by_dcos_uid_password_auth(
+            dcos_url, username, password)
+    if not dcos_token:
+        raise DCOSException("Authentication failed")
+    else:
+        return
+
+
+def oidc_implicit_flow_auth(dcos_url):
+    """
+    Authenticate user using OIDC implict flow
+
+    Raises exception if authentication fails.
+
+    :param dcos_url: url to cluster
+    :type dcos_url: str
+    :rtype: None
+    """
+
+    dcos_token = _get_dcostoken_by_oidc_implicit_flow(dcos_url)
+    if not dcos_token:
+        raise DCOSException("Authentication failed")
+    else:
+        return
+
+
+def _get_dcostoken_by_oidc_implicit_flow(dcos_url):
+    """
+    Get DC/OS Authentication Token by OIDC implicit flow
+
+    :param dcos_url: url to cluster
+    :type dcos_url: str
+    :returns: DC/OS Authentication Token
+    :rtype: str
+    """
+
+    oauth_login = 'login?redirect_uri=urn:ietf:wg:oauth:2.0:oob'
+    url = urllib.parse.urljoin(dcos_url, oauth_login)
+    oidc_token = _prompt_user_for_token(url, "OpenID Connect ID")
```

Comment:
"OpenID Connect ID Token", as in http://openid.net/specs/openid-connect-core-1_0.html#IDToken